app/questionnaire/questionnaire_manager.py

```diff
@@ -55,31 +55,33 @@ def validate_all_answers(self):
         answers = get_answers(current_user)
 
         for location in self.navigator.get_location_path(answers):
-            is_valid = self.validate(location, get_answers(current_user))
+            is_valid = self.validate(location, answers)
 
             if not is_valid:
                 logger.debug("Failed validation with current location %s", location)
                 return False, location
 
         return True, None
 
+    def update_questionnaire_store(self, location):
+        # Store answers in QuestionnaireStore
+        questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
+
+        for answer in self.get_state_answers(location):
```

Comment:
It is indeed, a previous comment by Andrew on one of my PRs confused me about this functionality.